examples/rapids/01-rapids-single-gpu.ipynb

```diff
@@ -0,0 +1,420 @@
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`parse_dates` tells `cudf` to interpret the \"tpep_pickup_datetime\" and \"tpep_dropoff_datetime\" columns in the datasets as datetime columns. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "***\n",
+    "Many dataframe operations that you would execute on a `pandas` dataframe, like `.head()` or `.dtypes`, also work for a `cudf` dataframe."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can compute the length and memory usage of the dataset using the following code."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "num_rows = len(taxi)\n",
+    "memory_usage = round(taxi.memory_usage(deep=True).sum() / 1e9, 2)\n",
+    "print(f\"Num rows: {num_rows}, Memory Usage: {memory_usage} GB\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Preprocess the Data\n",
+    "The raw data we downloaded needs to be processed before we can use it to train our machine learning model. We need to do things like create a target column, add additional features, and remove unnecessary columns. We will wrap everything in a function so we can use it later when we need to prepare data for testing or implementation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def prep_df(df: cudf.DataFrame) -> cudf.DataFrame:\n",
+    "\n",
+    "    df = df[df[\"fare_amount\"] > 0]  # to avoid a divide by zero error\n",
+    "    df[\"tip_fraction\"] = df[\"tip_amount\"] / df[\"fare_amount\"]\n",
+    "    df[\"target\"] = df[\"tip_fraction\"] > 0.2\n",
+    "\n",
+    "    df[\"pickup_weekday\"] = df[\"tpep_pickup_datetime\"].dt.weekday\n",
+    "    df[\"pickup_hour\"] = df[\"tpep_pickup_datetime\"].dt.hour\n",
+    "    df[\"pickup_week_hour\"] = (df[\"pickup_weekday\"] * 24) + df.pickup_hour\n",
+    "    df[\"pickup_minute\"] = df[\"tpep_pickup_datetime\"].dt.minute\n",
+    "\n",
+    "    df = df[\n",
+    "        [\n",
+    "            \"pickup_weekday\",\n",
+    "            \"pickup_hour\",\n",
+    "            \"pickup_week_hour\",\n",
+    "            \"pickup_minute\",\n",
+    "            \"passenger_count\",\n",
+    "            \"PULocationID\",\n",
+    "            \"DOLocationID\",\n",
+    "            \"target\",\n",
+    "        ]\n",
+    "    ]\n",
+    "\n",
+    "    df = df.astype(\"float32\").fillna(-1)\n",
+    "    df[\"target\"] = df[\"target\"].astype(\"int32\")\n",
+    "\n",
+    "    return df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<br>\n",
+    "<details><summary>If you want to learn more about how this function works, click here. </summary>\n",
```

Comment:
We could maybe remove this? I think it's kind of self evident but I'm not positive.